playground/AspireEventHub/EventHubs.AppHost/eventhubns.module.bicep

```diff
@@ -1,47 +1,36 @@
-targetScope = 'resourceGroup'
-
-@description('')
+@description('The location for the resource(s) to be deployed.')
 param location string = resourceGroup().location
 
-@description('')
 param sku string = 'Standard'
 
-@description('')
 param principalId string
 
-@description('')
 param principalType string
 
-
-resource eventHubsNamespace_wORIGuvCQ 'Microsoft.EventHub/namespaces@2021-11-01' = {
-  name: toLower(take('eventhubns${uniqueString(resourceGroup().id)}', 24))
-  location: location
-  tags: {
-    'aspire-resource-name': 'eventhubns'
-  }
-  sku: {
-    name: sku
-  }
-  properties: {
-  }
+resource eventhubns 'Microsoft.EventHub/namespaces@2017-04-01' = {
+    name: take('eventhubns-${uniqueString(resourceGroup().id)}', 256)
```

Comment:
Needs `toLower()`

Reply:
@tg-msft told me it wasn't necessary because `eventhubns-` is all lower, and `uniqueString` produces a lowered string.

Reply:
The resource name in this instance is going to be different. You are going from this:

```
toLower(take('eventhubns${uniqueString(resourceGroup().id)}', 24))
```

to this:

```
take('eventhubns-${uniqueString(resourceGroup().id)}', 256)
```

If you take the same code from Aspire 8.0 to Aspire 9.0 you are going to end up with an extra resource being provisioned. We need to be 100% sure the generated resource names align between Aspire 8.0 and Aspire 9.0, otherwise we are going to break people who have deployed to production.

In this particular case its the `-` between `eventhubns` and the generated unique string which will be a problem.

In the case of storage, the resource name MUST be lower case so there would be a deployment failure if someone named their resource in Aspire with an uppercase (this is why we ended up with `toLower(...)` in the first place in order to normalize names into the most restrictive subset of what Azure requires.

Reply:
The old implementation was being overly aggressive.  Now we only sanitize out characters that aren't allowed and will change case where possible (i.e., there's no `toLower` on Storage accounts either because we'll just generate a lowercase prefix).  I plan to remove the `take`s as well when the prefix + 13 is less than max so you get nicer looking bicep. 

If you all want the exact behavior from v8 though, you can implement a property resolver like https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/provisioning/Azure.Provisioning/src/Primitives/ResourceNamePropertyResolver.cs#L186 that toLowers dashless names.  Add it to your provisioning context when calling `.Build` and it'll change how everything is named for you.

Reply:
I've updated this PR to follow the existing naming scheme - https://github.com/dotnet/aspire/pull/5592/commits/16ea988c6b7b709fed3e3d1969ae36b2785254ee.

And I've logged https://github.com/dotnet/aspire/issues/5756 to decide on how to address the naming scheme going forward.